Basic/ClientDriven/Assets/Scripts/ServerIngredientSpawner.cs

```diff
@@ -27,19 +33,24 @@ public override void OnNetworkSpawn()
         }
     }
 
-    private void FixedUpdate()
+    void FixedUpdate()
     {
-        if (NetworkManager != null && !IsServer) return;
-        if (Time.time - m_LastSpawnTime > m_SpawnRatePerSecond)
+        if (NetworkManager != null && !IsServer)
+        {
+            return;
+        }
+
+        if (m_SpawnWaves < m_MaxSpawnWaves && Time.time - m_LastSpawnTime > m_SpawnRatePerSecond)
         {
             foreach (var spawnPoint in m_SpawnPoints)
             {
                 var newIngredientObject = Instantiate(m_IngredientPrefab, spawnPoint.transform.position, spawnPoint.transform.rotation);
                 newIngredientObject.transform.position = spawnPoint.transform.position;
                 var ingredient = newIngredientObject.GetComponent<ServerIngredient>();
-                ingredient.CurrentIngredientType.Value = (IngredientType) r.Next((int)IngredientType.max);
                 ingredient.NetworkObject.Spawn();
+                ingredient.CurrentIngredientType.Value = (IngredientType)m_RandomGenerator.Next((int)IngredientType.MAX);
             }
+            m_SpawnWaves++;
```

Comment:
I don't see spawnWaves being reset anywhere or decremented. Does it mean if players place all ingredients in the circles, there won't be new ingredients?
(That can work, I don't mind, just checking if this is the intended behaviour)